internal/pkg/template/upgrade/inputs.go

```diff
@@ -84,6 +84,9 @@ func (e inputsValuesExporter) addValue(key model.Key, content *orderedmap.Ordere
 		e.logger.Writef(`Value for input "%s" found, but type doesn't match, JSON key "%s", in %s`, inputId, jsonKey, key.Desc())
 		return
 	}
+	if inputDef.Type == input.TypeObject {
+		value = value.(*orderedmap.OrderedMap).ToMap()
+	}
```

Comment:
No a tady ten chybějící default. `content` je celý json dané konfigurace a na řádku 75 se z něj přes `content.GetNested()` vybere jen ta část, kterou tam vložil ten daný input. Jenže `content` a potom i ta `value` jsou typu `OrderedMap` což je taková struktura která nad mapou (obdoba asociativního pole v php) přidává zachování pořadí klíčů. 

Jenže input který má `type` `object` musí mít typ `map`. To se na řádku 90 zavolá `inputDef.Type.ValidateValue()` a tady to potom vyfailuje:
https://github.com/keboola/keboola-as-code/blob/021aacd4a89ba7487cc46f1d2ce22551453f4391/internal/pkg/template/input/type.go#L114-L116
protože `valueKind` je `"OrderedMap"` ale my tam očekáváme `reflect.Map` (tj. `"map"`).

Zkrátka, **TLDR** , tady musím udělat to, že pokud je `type` inputu `object`, tak tu `OrderedMap` ve value převedeme na `map` pomocnou metodou `.ToMap()`. 🙂 Nejlepší když se koukneš v dalších commitech na E2E test. 😃

Reply:
Tomuhle výjimečně rozumím. V JS používáme taky `Map` a `OrderedMap` :D